### PR TITLE
Fix hard-coded paths in skimmer

### LINKED ISSUE #32: Hardcoded paths in produce.cc
Hardcoded input directory in `produce.cc`
https://github.com/bfonta/bye_splits/blob/ff9e6ec554c89ef70038cc4d48460d960bf3f76e/bye_splits/production/produce.cc#L78
While we should expect it to be read from the configuration file
https://github.com/bfonta/bye_splits/blob/ff9e6ec554c89ef70038cc4d48460d960bf3f76e/bye_splits/production/prod_params.yaml#L4-L6

In addition,  at the moment it can only use the same base directory for input and output files:
https://github.com/bfonta/bye_splits/blob/ff9e6ec554c89ef70038cc4d48460d960bf3f76e/bye_splits/production/produce.cc#L92

But the input directory could be read-only (for instance if reading data in another user directory), so it would be better to have a separate base directory to store the output. And ideally this output base directory should be configurable as well.

### PR DESCRIPTION
**Pull request by bfonta**
This PR fixes #32. Tested.

### CODE REVIEW

**Review by isehle — APPROVED** [review]
Looks good, thanks!

**Review by mchiusi — COMMENTED** [review]
Thank you for fixing the hard-coded paths. It would be nice if, in the near future, we could automatically change the output file location. Currently, it is hardcoded in the config file, but it would be better if the path could depend on the user running the code and on an output-folder that could be passed via the parser.

File: config.yaml [hunk 1]
```diff
@@ -1,4 +1,20 @@
 io:
+  production:
+    photons:
+      dir: FloatingpointMixedbcstcrealsig4DummyHistomaxxydr015GenmatchGenclustersntuple
+      tree: HGCalTriggerNtuple
+      infile: /eos/user/b/bfontana/FPGAs/new_algos/photons_0PU_bc_stc_hadd.root
+      outfile: skim_photons_default.root
```
[the remark this answers is not in the document]
Comment (bfonta) [reply config.yaml:7]: I don't; it is a default that a user can change. 
Moreover, there are occasionally some I/O errors when copying files to ```/eos/```, so this at least likely runs during a local test.

**Comment by bfonta** [comment]
@mchiusi 

> It would be nice if, in the near future, we could automatically change the output file location

I definitely agree, but I don't know of any ideal way to do it. We can discuss.

**Review by mchiusi — APPROVED** [review]

**Pull request merged**